### PR TITLE
Use assertj fluent style

### PR DESCRIPTION
**Pull request by dbmalkovsky**
I noticed in some recent pushes some files with a mix of assertion styles.   It appeared that the Assertj Fluent style was the newer preferred format so I started updating the mixed files and then expanded to the entire package level.   

If this is acceptable I will continue to make similar changes in other packages time permitting.

### CODE REVIEW

**Comment by dbmalkovsky** [comment]
The tests passed in my Travis CI build; see https://travis-ci.org/dbmalkovsky/flowable-engine/builds/653900593

File: modules/flowable-form-spring/src/test/java/org/flowable/spring/test/autodeployment/SpringAutoDeployTest.java [hunk 1]
```diff
@@ -134,34 +132,34 @@ public void testBasicFlowableSpringIntegration() {
         expectedFormDefinitionKeys.add("form1");
         expectedFormDefinitionKeys.add("form2");
 
-        assertEquals(expectedFormDefinitionKeys, formDefinitionKeys);
+        assertThat(formDefinitionKeys).isEqualTo(expectedFormDefinitionKeys);
```
Comment (filiphr) [inline modules/flowable-form-spring/src/test/java/org/flowable/spring/test/autodeployment/SpringAutoDeployTest.java]: We can get rid of the loop and expected form definition keys by using:

```
assertThat(formDefinitions)
    .extracting(FormDefinition::getKey)
    .containsExactlyInAnyOrder("form1", "form2");
```

File: modules/flowable-app-engine/src/test/java/org/flowable/app/engine/test/repository/CaseDefinitionQueryTest.java [hunk 2]
```diff
@@ -82,59 +81,60 @@ public void deleteTestDeployments() {
...
-        assertEquals(0, appRepositoryService.createAppDefinitionQuery().deploymentIds(new HashSet<>(Collections.singletonList("invalid"))).list().size());
-        assertEquals(0, appRepositoryService.createAppDefinitionQuery().deploymentIds(new HashSet<>(Collections.singletonList("invalid"))).count());
+        assertThat(appRepositoryService.createAppDefinitionQuery().deploymentIds(new HashSet<>(Collections.singletonList("invalid"))).list()).hasSize(0);
+        assertThat(appRepositoryService.createAppDefinitionQuery().deploymentIds(new HashSet<>(Collections.singletonList("invalid"))).count()).isEqualTo(0);
     }
 
     @Test
     public void testQueryByEmptyDeploymentIds() {
-        try {
+        assertThrows(FlowableIllegalArgumentException.class, () -> {
```
[the remark this answers is not in the document]
Comment (dbmalkovsky) [reply modules/flowable-app-engine/src/test/java/org/flowable/app/engine/test/repository/CaseDefinitionQueryTest.java]: Thanks; done.

File: modules/flowable-app-engine/src/test/java/org/flowable/app/engine/test/repository/CaseDefinitionQueryTest.java [hunk 3]
```diff
@@ -143,325 +143,326 @@ public void testQueryByAppDefinitionId() {
...
+        assertThat(appRepositoryService.createAppDefinitionQuery().orderByAppDefinitionCategory().desc().count()).isEqualTo(4);
     }
 
     @Test
     public void testQueryOrderByCaseDefinitionKey() {
-        assertEquals(4, appRepositoryService.createAppDefinitionQuery().orderByAppDefinitionKey().asc().list().size());
-        assertEquals(4, appRepositoryService.createAppDefinitionQuery().orderByAppDefinitionKey().asc().count());
+        assertThat(appRepositoryService.createAppDefinitionQuery().orderByAppDefinitionKey().asc().list()).hasSize(4);
+        assertThat(appRepositoryService.createAppDefinitionQuery().orderByAppDefinitionKey().asc().count()).isEqualTo(4);
         List<AppDefinition> appDefinitions = appRepositoryService.createAppDefinitionQuery().orderByAppDefinitionKey().asc().list();
```
[the remark this answers is not in the document]
Comment (dbmalkovsky) [reply modules/flowable-app-engine/src/test/java/org/flowable/app/engine/test/repository/CaseDefinitionQueryTest.java]: Thanks; done.

File: modules/flowable-app-engine/src/test/java/org/flowable/app/engine/test/repository/CaseDefinitionQueryTest.java [hunk 4]
```diff
@@ -143,325 +143,326 @@ public void testQueryByAppDefinitionId() {
...
-                assertEquals("fullInfoApp", appDefinitions.get(i).getKey());
+                assertThat(appDefinitions.get(i).getKey()).isEqualTo("fullInfoApp");
             }
         }
 
-        assertEquals(4, appRepositoryService.createAppDefinitionQuery().orderByAppDefinitionKey().desc().list().size());
-        assertEquals(4, appRepositoryService.createAppDefinitionQuery().orderByAppDefinitionKey().desc().count());
+        assertThat(appRepositoryService.createAppDefinitionQuery().orderByAppDefinitionKey().desc().list()).hasSize(4);
+        assertThat(appRepositoryService.createAppDefinitionQuery().orderByAppDefinitionKey().desc().count()).isEqualTo(4);
         appDefinitions = appRepositoryService.createAppDefinitionQuery().orderByAppDefinitionKey().desc().list();
```
[the remark this answers is not in the document]
Comment (dbmalkovsky) [reply modules/flowable-app-engine/src/test/java/org/flowable/app/engine/test/repository/CaseDefinitionQueryTest.java]: Thanks; done.

File: modules/flowable-app-engine/src/test/java/org/flowable/app/engine/test/repository/CaseDefinitionQueryTest.java [hunk 5]
```diff
@@ -143,325 +143,326 @@ public void testQueryByAppDefinitionId() {
...
+        assertThat(appDefinitions.get(1).getVersion()).isEqualTo(2);
+        assertThat(appDefinitions.get(2).getVersion()).isEqualTo(1);
+        assertThat(appDefinitions.get(3).getVersion()).isEqualTo(1);
 
         appDefinitions = appRepositoryService.createAppDefinitionQuery().latestVersion().orderByAppDefinitionVersion().asc().list();
-        assertEquals(1, appDefinitions.get(0).getVersion());
-        assertEquals("fullInfoApp", appDefinitions.get(0).getKey());
-        assertEquals(3, appDefinitions.get(1).getVersion());
-        assertEquals("testApp", appDefinitions.get(1).getKey());
+        assertThat(appDefinitions.get(0).getVersion()).isEqualTo(1);
```
[the remark this answers is not in the document]
Comment (dbmalkovsky) [reply modules/flowable-app-engine/src/test/java/org/flowable/app/engine/test/repository/CaseDefinitionQueryTest.java]: Thanks; done.

File: modules/flowable-app-engine/src/test/java/org/flowable/app/engine/test/repository/CustomAppModelTest.java [hunk 6]
```diff
@@ -37,18 +35,18 @@ public void testAppDefinitionDeployed() throws Exception {
...
-            assertNotNull(appDeployment);
+            assertThat(appDeployment).isNotNull();
             
             AppDefinition appDefinition = appRepositoryService.createAppDefinitionQuery().appDefinitionKey("extraInfoApp").singleResult();
             
             AppModel appModel = appRepositoryService.getAppModel(appDefinition.getId());
-            assertNotNull(appModel);
-            assertTrue(appModel instanceof CustomAppModel);
+            assertThat(appModel).isNotNull();
+            assertThat(appModel instanceof CustomAppModel).isTrue();
```
[the remark this answers is not in the document]
Comment (dbmalkovsky) [reply modules/flowable-app-engine/src/test/java/org/flowable/app/engine/test/repository/CustomAppModelTest.java]: Thanks; done.

File: modules/flowable-app-engine/src/test/java/org/flowable/app/engine/test/repository/DeploymentQueryTest.java [hunk 7]
```diff
@@ -69,103 +66,104 @@ public void testQueryNoParams() {
                 deployment2Found = true;
             }
         }
-        assertTrue(deployment1Found);
-        assertTrue(deployment2Found);
+        assertThat(deployment1Found).isTrue();
```
[the remark this answers is not in the document]
Comment (dbmalkovsky) [reply modules/flowable-app-engine/src/test/java/org/flowable/app/engine/test/repository/DeploymentQueryTest.java]: Thanks; done.

File: modules/flowable-app-engine/src/test/java/org/flowable/app/engine/test/repository/DeploymentTest.java [hunk 8]
```diff
@@ -49,53 +46,53 @@
...
     public void testAppDefinitionDeployed() throws Exception {
         org.flowable.app.api.repository.AppDeployment appDeployment = appRepositoryService.createDeploymentQuery().singleResult();
-        assertNotNull(appDeployment);
+        assertThat(appDeployment).isNotNull();
         
         List<String> resourceNames = appRepositoryService.getDeploymentResourceNames(appDeployment.getId());
-        assertEquals(1, resourceNames.size());
-        assertEquals("org/flowable/app/engine/test/repository/DeploymentTest.testAppDefinitionDeployed.app", resourceNames.get(0));
+        assertThat(resourceNames.size()).isEqualTo(1);
+        assertThat(resourceNames.get(0)).isEqualTo("org/flowable/app/engine/test/repository/DeploymentTest.testAppDefinitionDeployed.app");
```
[the remark this answers is not in the document]
Comment (dbmalkovsky) [reply modules/flowable-app-engine/src/test/java/org/flowable/app/engine/test/repository/DeploymentTest.java]: Thanks; done.

File: modules/flowable-app-engine/src/test/java/org/flowable/app/engine/test/repository/DeploymentTest.java [hunk 9]
```diff
@@ -104,38 +101,38 @@ public void testAppDefinitionZipDeployed() throws Exception {
...
-        assertNotNull(appDefinition.getId());
-        assertNotNull(appDefinition.getDeploymentId());
-        assertNotNull(appDefinition.getKey());
-        assertNotNull(appDefinition.getResourceName());
-        assertTrue(appDefinition.getVersion() > 0);
+        assertThat(appDefinition.getId()).isNotNull();
+        assertThat(appDefinition.getDeploymentId()).isNotNull();
+        assertThat(appDefinition.getKey()).isNotNull();
+        assertThat(appDefinition.getResourceName()).isNotNull();
+        assertThat(appDefinition.getVersion() > 0).isTrue();
```
[the remark this answers is not in the document]
Comment (dbmalkovsky) [reply modules/flowable-app-engine/src/test/java/org/flowable/app/engine/test/repository/DeploymentTest.java]: Thanks; done.

File: modules/flowable-event-registry-spring/src/test/java/org/flowable/eventregistry/spring/test/autodeployment/SpringAutoDeployTest.java [hunk 10]
```diff
@@ -125,34 +123,34 @@ public void testBasicFlowableSpringIntegration() {
...
         createAppContextWithoutDeploymentMode();
-        assertEquals(1, repositoryService.createDeploymentQuery().count());
+        assertThat(repositoryService.createDeploymentQuery().count()).isOne();
         applicationContext.close();
 
         String filePath = "org/flowable/eventregistry/spring/test/autodeployment/simpleEvent.event";
         String originalEventFileContent = IoUtil.readFileAsString(filePath);
         String updatedEventFileContent = originalEventFileContent.replace("My event", "My event test");
-        assertTrue(updatedEventFileContent.length() > originalEventFileContent.length());
+        assertThat(updatedEventFileContent.length() > originalEventFileContent.length()).isTrue();
```
[the remark this answers is not in the document]
Comment (dbmalkovsky) [reply modules/flowable-event-registry-spring/src/test/java/org/flowable/eventregistry/spring/test/autodeployment/SpringAutoDeployTest.java]: Thanks; done.

File: modules/flowable-form-engine/src/test/java/org/flowable/form/engine/test/DeploymentTest.java [hunk 11]
```diff
@@ -43,15 +42,15 @@ public void redeploySingleForm() {
...
+        assertThat(formDefinition.getVersion()).isOne();
 
         FormInfo formInfo = repositoryService.getFormModelByKey("form1");
         SimpleFormModel formModel = (SimpleFormModel) formInfo.getFormModel();
-        assertEquals(1, formModel.getFields().size());
-        assertEquals("input1", formModel.getFields().get(0).getId());
-        assertEquals("Input1", formModel.getFields().get(0).getName());
+        assertThat(formModel.getFields().size()).isOne();
+        assertThat(formModel.getFields().get(0).getId()).isEqualTo("input1");
+        assertThat(formModel.getFields().get(0).getName()).isEqualTo("Input1");
```
[the remark this answers is not in the document]
Comment (dbmalkovsky) [reply modules/flowable-form-engine/src/test/java/org/flowable/form/engine/test/DeploymentTest.java]: Thanks; done.

File: modules/flowable-form-engine/src/test/java/org/flowable/form/engine/test/DeploymentTest.java [hunk 12]
```diff
@@ -61,15 +60,15 @@ public void redeploySingleForm() {
...
+        assertThat(formDefinition.getVersion()).isEqualTo(2);
 
         formInfo = repositoryService.getFormModelByKey("form1");
         formModel = (SimpleFormModel) formInfo.getFormModel();
-        assertEquals(1, formModel.getFields().size());
-        assertEquals("input2", formModel.getFields().get(0).getId());
-        assertEquals("Input2", formModel.getFields().get(0).getName());
+        assertThat(formModel.getFields().size()).isOne();
+        assertThat(formModel.getFields().get(0).getId()).isEqualTo("input2");
+        assertThat(formModel.getFields().get(0).getName()).isEqualTo("Input2");
```
[the remark this answers is not in the document]
Comment (dbmalkovsky) [reply modules/flowable-form-engine/src/test/java/org/flowable/form/engine/test/DeploymentTest.java]: Thanks; done.

File: modules/flowable-form-engine/src/test/java/org/flowable/form/engine/test/DeploymentTest.java [hunk 13]
```diff
@@ -79,10 +78,10 @@ public void redeploySingleForm() {
             "org/flowable/form/engine/test/deployment/form_with_dates.form" })
     public void deploy2Forms() {
         List<FormDefinition> formDefinitions = repositoryService.createFormDefinitionQuery().orderByFormName().asc().list();
-        assertEquals(2, formDefinitions.size());
+        assertThat(formDefinitions.size()).isEqualTo(2);
 
-        assertEquals("My date form", formDefinitions.get(0).getName());
-        assertEquals("My first form", formDefinitions.get(1).getName());
+        assertThat(formDefinitions.get(0).getName()).isEqualTo("My date form");
+        assertThat(formDefinitions.get(1).getName()).isEqualTo("My first form");
```
[the remark this answers is not in the document]
Comment (dbmalkovsky) [reply modules/flowable-form-engine/src/test/java/org/flowable/form/engine/test/DeploymentTest.java]: Thanks; done.

File: modules/flowable-form-engine/src/test/java/org/flowable/form/engine/test/ForceCloseMybatisConnectionPoolTest.java [hunk 14]
```diff
@@ -41,14 +40,14 @@ public void testForceCloseMybatisConnectionPoolTrue() {
 
         PooledDataSource pooledDataSource = (PooledDataSource) standaloneInMemFormEngineConfiguration.getDataSource();
         PoolState state = pooledDataSource.getPoolState();
-        assertTrue(state.getIdleConnectionCount() > 0);
+        assertThat(state.getIdleConnectionCount()).isGreaterThan(0);
```
[the remark this answers is not in the document]
Comment (dbmalkovsky) [reply modules/flowable-form-engine/src/test/java/org/flowable/form/engine/test/ForceCloseMybatisConnectionPoolTest.java]: Thanks; done.

File: modules/flowable-form-engine/src/test/java/org/flowable/form/engine/test/FormInstanceTest.java [hunk 15]
```diff
@@ -51,23 +48,23 @@ public void submitSimpleForm() throws Exception {
...
+        assertThat(formInstanceModel.getKey()).isEqualTo("form1");
         
         SimpleFormModel formModel = (SimpleFormModel) formInstanceModel.getFormModel();
-        assertEquals(1, formModel.getFields().size());
+        assertThat(formModel.getFields().size()).isOne();
         FormField formField = formModel.getFields().get(0);
-        assertEquals("input1", formField.getId());
-        assertEquals("test", formField.getValue());
+        assertThat(formField.getId()).isEqualTo("input1");
+        assertThat(formField.getValue()).isEqualTo("test");
```
[the remark this answers is not in the document]
Comment (dbmalkovsky) [reply modules/flowable-form-engine/src/test/java/org/flowable/form/engine/test/FormInstanceTest.java]: Thanks; done.

File: modules/flowable-form-spring/src/test/java/org/flowable/spring/test/autodeployment/SpringAutoDeployTest.java (hunk 1 above)
Comment (dbmalkovsky) [reply modules/flowable-form-spring/src/test/java/org/flowable/spring/test/autodeployment/SpringAutoDeployTest.java]: Thanks; done.

**Comment by dbmalkovsky** [comment]
@filiphr Thanks for the review; I've made the changes and found a few more instances and files to change in the same packages.

File: modules/flowable-app-engine/src/test/java/org/flowable/app/engine/test/repository/DeploymentQueryTest.java [hunk 16]
```diff
@@ -57,115 +54,108 @@ public void deleteTestDeployments() {
...
-            }
-        }
-        assertTrue(deployment1Found);
-        assertTrue(deployment2Found);
+        assertThat(appRepositoryService.createDeploymentQuery().list()).hasSize(2);
+        assertThat(appRepositoryService.createDeploymentQuery().count()).isEqualTo(2);
+
+        assertThat(appRepositoryService.createDeploymentQuery().list())
+            .extracting(AppDeployment::getId)
+            .containsExactly(deploymentId1, deploymentId2);
```
Comment (filiphr) [inline modules/flowable-app-engine/src/test/java/org/flowable/app/engine/test/repository/DeploymentQueryTest.java]:
```suggestion
            .containsExactlyInAnyOrder(deploymentId1, deploymentId2);
```

**Comment by filiphr** [comment]
Thanks for the PR David. 

>If this is acceptable I will continue to make similar changes in other packages time permitting.

I would say that it is. I would only suggest to do it package by package in different PRs. Especially for the CMMN and Process Engine. Otherwise it would be really difficult to review all of it at once :)

**Pull request merged**